src/test/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlightTest.java

```diff
@@ -44,25 +44,28 @@ void beforeEach() {
     when(appConfig.getMaxStairwayThreads()).thenReturn(1);
 
     SnapshotService snapshotService = mock(SnapshotService.class);
-    DatasetSummary datasetSummary = mock(DatasetSummary.class);
-    when(snapshotService.getSourceDatasetsFromSnapshotRequest(any()))
-        .thenReturn(List.of(new Dataset(datasetSummary)));
+    DatasetService datasetService = mock(DatasetService.class);
+    UUID datasetId = UUID.randomUUID();
+    when(datasetService.retrieve(datasetId)).thenReturn(mock(Dataset.class));
 
     when(context.getBean(any(Class.class))).thenReturn(null);
     when(context.getBean(anyString(), any(Class.class))).thenReturn(null);
     // Beans that are interacted with directly in flight construction rather than simply passed
     // to steps need to be added to our context mock.
     when(context.getBean(ApplicationConfiguration.class)).thenReturn(appConfig);
     when(context.getBean(SnapshotService.class)).thenReturn(snapshotService);
+    when(context.getBean(DatasetService.class)).thenReturn(datasetService);
 
     inputParameters = new FlightMap();
+    // Because this isn't a swagger type, this can't be converted with the objectmapper
```

Comment:
this meaning the request? But it is put in the map?

Reply:
Ah, that's outdated. Thank you.